Final Project Data Set: Phylogeny of Gonioctena leaf beetles (Coleoptera; Chrysomelidae) 

"A phylogenetic analysis of the genus Gonioctena (Coleoptera, Chrysomelidae) based on allozyme data (17 loci) and mitochondrial DNA sequence data (three gene fragments, 1,391 sites) was performed to study the evolutionary history of host-plant shifts among these leaf beetles. This chrysomelid genus is characteristically associated with a high number of different plant families. The diverse molecular data gathered in this study are to a large extent congruent, and the analyses provide a well-supported phylogenetic hypothesis to address questions about the evolution of host-plant shifts in the genus Gonioctena. The most-parsimonious reconstruction of the ancestral host-plant associations, based on the estimated phylogeny, suggests that the Fabaceae was the ancestral host-plant family of the genus. Although most of the host-plant shifts (between different host species) in Gonioctena have occurred within the same plant family or within the same plant genus, at least eight shifts have occurred between hosts belonging to distantly related and chemically dissimilar plant families." The link to my data set is copied below: 
https://pubmed.ncbi.nlm.nih.gov/11975339/

SEQUENCE ALIGNMENT USING CLUSALW: 

I attemped to download miniconda (https://docs.conda.io/en/latest/miniconda.html), but was unable to initially. After a little while, I was able to install miniconda and use it to install ClusalW. Once miniconda was downloaded, I used the code "conda install -c bioconda clustalw" to install ClusalW. 

Additionally, I downloaded MAFFT (https://mafft.cbrc.jp/alignment/software/).

I copied and pasted the sequences for each gene marker (from my data) into a Visual Studio Code-3. This process took some time as my data set would not let me copy and paste every sequecne at the same time. When reproducing this project, I would take into account that this process takes a lot of time to complete. When I finished copying my data set completely, I saved each one as a text file. I saved each text file in my Botany-563 folder on my Desktop. I named each folder according to which marker they contained. I named the first text file "beetles-COII-data", the second file "beetles-12s-data", and the third file "beetles-16s-data." Remember to convert all of our data files into a .txt file. 

Then, I realized that the data sequences from my data set are already aligned, and that I did not need to realign them. From the article, I learned that the researchers aligned the data sequences with ClusalW, which I would have used as well if the data was not already aligned. 

DISTANCE AND PARSIMONY USING R: 

I downloaded R studio and R using this link: https://posit.co/download/rstudio-desktop/. In R Studio, I installed the various packages by using the code "install.packages("adegenet", dep=TRUE)" and then using the code "install.packages("phangorn", dep=TRUE)". After installing the packages, I used the code "library(ape)", then  "library(adegenet)", and finally "library(phangorn)".

I tried to create a tree on R, but was unsuccessful. I got many errors when I used the code "dna <- fasta2DNAbin(file="beetles_COII_data.fasta")". If this code were successful, then you would use the code "D <- dist.dna(dna, model="TN93")", then you would use the code "tre <- nj(D)", and finally "plot(tre).  

After struggling to make a phylogenetic tree on R, I decided to go to office hours and ask for Claudia's assistance. She helped me create the tree on R using these codes:"dna <- fasta2DNAbin(file="beetles_COII_data.fasta")", then "D <- dist.dna(dna, model="TN93")", then you would use the code "tre <- nj(D)", and finally "plot(tre). This time, I was successful in making three trees for each of my geentic markers, COII, 12s, and 16s. With the help of Claudia, I realized that I had my DNA sequences in the wrong format. I had to put all DNA sequences for each species on one, single line in Visual Studio Code. After manipulating my data, I had no trouble creating each phylogenetic tree. 

MAXIMUM LIKELIHOOD USING IQTREE:

I downloaded iqtree using this link: http://www.iqtree.org/#download. I made sure that I was in my Botany-563 folder before I ran the code to create my maximum likelihood tree. After making sure I was in the right folder, I then ran the command "./iqtree -s beetles_COII_data.fasta -bb 1000 -nt AUTO". This command selects the correct alignment file: beetles_COII_data.fasta, while also specifying "1000 replicates for the ultrafast bootstrap" with the command "-bb 1000". Additionally, the code "-nt AUTO" is used "to determine the best number of CPU cores to speed up the analysis". It only took about 30 seconds to create each maximum likelihood tree and other corresponding folders for each data set. To view the tree, I exported each tree file (beetles_COII_data.fasta.treefile)to R. In the terminal I pressed "R" and enter to run R in the terminal. Then, I ran the code "library (ape)" to import the correct packages to view my maximum likelihood tree. After importing this package, I ran the code "tre <- read.tree(text="(G_viminalis_1:0.0217072984,((((((G_Olivacea_1:0.1520952538,G_variabilis_1:0.1125003548)72:0.0310592011,(G_fornicata:0.0871064650,(G_nigropiagiata:0.1197199436,Oreina_cacaiiae:0.6011861054)37:0.0718537521)29:0.0362086263)46:0.0341164786,(((((G_interposita_1:0.0051242681,G_interposita_2:0.0059512108)85:0.0093012210,(G_pallida_2:0.0203925877,G_pallida_3:0.0042111541)88:0.0073092556)97:0.0593812674,(G_intermedia:0.0340313171,G_quinquepunctata_2:0.0559322314)87:0.0297177647)93:0.0776327971,Chrysomela_tremula:0.5993413139)23:0.0206636333,(G_tredecimmaculata:0.1676951772,(G_kamikawai:0.1862538319,G_rubripennis:0.1496203046)73:0.0318662003)61:0.0296163881)36:0.0269980397)96:0.0658828058,G_linnaeana:0.1174307502)95:0.0278105299,G_rufipes:0.0735009840)94:0.0283487528,G_occidentalis:0.0466238825)95:0.0264197918,G_holdausi:0.0424594138);"). After running this code, I ran the code "plot(tre) to visualize the maximum likelihood tree. After succesfully creating this tree, I saved each tree into my Botany-563 folder. I have attched one of my maximum likelihood trees: <img width="552" alt="Maximum_Likelihood_Tree_IQTREE_beetles_COII_data" src="https://user-images.githubusercontent.com/123671543/236015418-1195e5e3-38d9-4b73-92b3-09e42a4580e2.png">

WHY I CHOSE TO USE IQTREE: 

I chose to use IQtree because it "has a number of features that reduce computational time (Ultrafast Bootstrap approximation, improved quartet likelihood mapping, etc)." Additionally, "IQTree samples several starting trees and randomly perturbs trees during the hill climbing process to prevent the ML hill-climbing algorithm from getting 'stuck' in local optima." I chose to use IQtree because it "uses a fast NNI hill climbing algorithm, faster than traditional NNI hill climbing algorithms, implements non-time-reversible models, allowing for inference of rooted phylogenies, and is highly versatile with lots of additional functionality." Overall, IQtree is extremely fast and mostly accurate (when using the optimal amount of data). However, depending on the amount of data inputted into IQtree (extremely small or extremely large amounts of data), it may not always create the most accurate tree. 






