Final Project Data Set: Phylogeny of Gonioctena leaf beetles (Coleoptera; Chrysomelidae):

To study the evolutionary history of host-plant shifts among these leaf beetles, a phylogenetic analysis of the genus, Gonioctena (Coleoptera, Chrysomelidae), based on allozyme data was conducted. The various DNA sequences used to create phylogenetic and maximum likelihood trees provide evidence and an explanation of the evolution of host-plant switches in the genus, Gonioctena. In order to learn more about the host-plant shifts in this genus, I used the aligned DNA sequences of 20 different species (G. viminalis (1), G. Olivacea (1), G. rufipes, G. linnaeana, G. variabilis (1), G. fornicata, G. occidentalis, G. interposita (l), G.interposita (2), G. intermedia, G. quinquepunctata (2), G. holdausi, G. pallida (2), G. pallida (3), G. tredecimmaculata, G. nigropiagiata, G. kamikawai, G. rubripennis, Oreina cacaiiae, chrysomela tremula) within the genus to create phylogenetic trees that display their relation to one another. Based on these distance and parsimony trees, maximum likelihood trees, and additional research conducted, it is evident that the Fabaceae is the Gonioctena’s host-plant family. In addition, the analysis reveals that multiple host-plant shifts have occurred within the genus, Gonioctena. The link to my data set is copied here: https://pubmed.ncbi.nlm.nih.gov/11975339/

SEQUENCE ALIGNMENT USING CLUSAL W:

I attempted to download miniconda (https://docs.conda.io/en/latest/miniconda.html), but was unable to initially. After a little while, I was able to install miniconda and use it to install Clustal W. Once miniconda was downloaded, I used the code "conda install -c bioconda clustalw" to install Clustal W.

Additionally, I downloaded MAFFT (https://mafft.cbrc.jp/alignment/software/).

I copied and pasted the sequences for each gene marker (from my data) into a Visual Studio Code-3. This process took some time as my data set would not let me copy and paste every sequence at the same time. When reproducing this project, I would take into account that this process takes a lot of time to complete. When I finished copying my data set completely, I saved each one as a text file. I saved each text file in my Botany-563 folder on my Desktop. I named each folder according to which marker they contained. I named the first text file "beetles-COII-data", the second file "beetles-12s-data", and the third file "beetles-16s-data." Remember to convert all of our data files into a .txt file. 

Then, I realized that the data sequences from my data set are already aligned, and that I did not need to realign them. From the article, I learned that the researchers aligned the data sequences with ClustalW, which I would have used as well if the data was not already aligned.

DISTANCE AND PARSIMONY USING R:

Using the aligned DNA sequences, I decided to make a distance and parsimony tree using R studio to account for the estimated evolutionary distances and the minimum number of changes in the DNA. One reason why I chose to use R studio to create these distance and parsimony trees is because it offers its users various options to choose from to create the phylogenetic tree that best fits their data. For instance, it provides us with many algorithms to choose from when creating our distance and parsimony tree.

To create these trees, I downloaded R studio and R using this link: https://posit.co/download/rstudio-desktop/. In R Studio, I installed the various packages by using the code "install.packages("adegenet", dep=TRUE)" and then using the code "install.packages("phangorn", dep=TRUE)". After installing the packages, I used the code "library(ape)", then "library(adegenet)", and finally "library(phangorn)". I tried to create a tree on R, but was unsuccessful. I got many errors when I used the code "dna <- fasta2DNAbin(file="beetles_COII_data.fasta")".

After struggling to make a distance and parsimony tree on R, I decided to go to office hours and ask for Professor Solis Lemus’s assistance. She helped me create the tree on R using these codes:"dna <- fasta2DNAbin(file="beetles_COII_data.fasta")", then "D <- dist.dna(dna, model="TN93")", then the code "tre <- nj(D)", and finally "plot(tre). For each of my distance and parsimony trees, I had to input the correct file names for each marker (COII, 12s, and 16s) into my code to ensure I was creating the correct trees. I was successful in making three trees for each of my genetic markers, COII, 12s, and 16s. With the help of Professor Solis Lemis, I realized that I had my DNA sequences in the wrong format previously. I had to put all DNA sequences for each species on one, single line in Visual Studio Code. After manipulating my data, I had no trouble creating each distance and parsimony tree.

WHY I CHOSE TO USE R STUDIO:

I chose to use R studio to create the distance and parsimony trees because it is extremely fast and flexible. Additionally, I chose to use R studio to create these distance and parsimony trees because it has a wide variety of packages. However, creating phylogenetic trees on R studio does have some disadvantages. Sometimes, the tree produced by R is inaccurate due to choosing the wrong algorithm for the data set. However, this is a user error, so R studio is very effective and accurate when creating distance and parsimony trees when executed correctly.

MAXIMUM LIKELIHOOD USING IQTREE:

I downloaded iqtree using this link: http://www.iqtree.org/#download. I made sure that I was in my Botany-563 folder before I ran the code to create my maximum likelihood tree. After making sure I was in the right folder, I then ran the command "./iqtree -s beetles_COII_data.fasta -bb 1000 -nt AUTO". This command selects the correct alignment file: beetles_COII_data.fasta, while also specifying "1000 replicates for the ultrafast bootstrap" with the command "-bb 1000". Additionally, the code "-nt AUTO" is used "to determine the best number of CPU cores to speed up the analysis". It only took about 30 seconds to create each maximum likelihood tree and other corresponding folders for each data set. To view the tree, I exported each tree file (beetles_COII_data.fasta.treefile)to R. In the terminal I pressed "R" and enter to run R in the terminal. Then, I ran the code "library (ape)" to import the correct packages to view my maximum likelihood tree. After importing this package, I ran the code "tre <- read.tree(text="(G_viminalis_1:0.0217072984,((((((G_Olivacea_1:0.1520952538,G_variabilis_1:0.1125003548)72:0.0310592011,(G_fornicata:0.0871064650,(G_nigropiagiata:0.1197199436,Oreina_cacaiiae:0.6011861054)37:0.0718537521)29:0.0362086263)46:0.0341164786,(((((G_interposita_1:0.0051242681,G_interposita_2:0.0059512108)85:0.0093012210,(G_pallida_2:0.0203925877,G_pallida_3:0.0042111541)88:0.0073092556)97:0.0593812674,(G_intermedia:0.0340313171,G_quinquepunctata_2:0.0559322314)87:0.0297177647)93:0.0776327971,Chrysomela_tremula:0.5993413139)23:0.0206636333,(G_tredecimmaculata:0.1676951772,(G_kamikawai:0.1862538319,G_rubripennis:0.1496203046)73:0.0318662003)61:0.0296163881)36:0.0269980397)96:0.0658828058,G_linnaeana:0.1174307502)95:0.0278105299,G_rufipes:0.0735009840)94:0.0283487528,G_occidentalis:0.0466238825)95:0.0264197918,G_holdausi:0.0424594138);"). After running this code, I ran the code "plot(tre) to visualize the maximum likelihood tree. After succesfully creating this tree, I saved each tree into my Botany-563 folder. I have attched one of my maximum likelihood trees: <img width="552" alt="Maximum_Likelihood_Tree_IQTREE_beetles_COII_data" src="https://github.com/aidynkehrli/Botany-563/assets/123671543/242c13c5-0753-4f13-ae0b-fe6cb5575a0c">

WHY I CHOSE TO USE IQTREE:

I chose to use IQtree to create my maximum likelihood trees because it has many strengths/advantages. IQTree has numerous features that decrease computational time. Some of these factors include ultrafast bootstrap approximation, improved quartet likelihood mapping, and so much more. Additionally, IQTree uses “a combination of hill-climbing approaches and a stochastic perturbation method that can be time-efficiently implemented” (Lam-Tung Nguyen et al., 2014). Another strength of the latest version of IQTree is that it “allows users to reconstruct rooted trees using non-reversible models, a feature not available in most ML packages due to numerical and computational expense” (Bui Quang Minh et al., 2020). I chose to use IQTree because it utilizes a fast NNI hill-climbing algorithm and implements non-time-reversible models that allow researchers to reconstruct rooted phylogenies. Additionally, IQTree allows for the users to choose what is best for their data. For instance, users are allowed to choose their preferred substitution model, their use of ultrafast bootstrapping or not, and if they want a “partition file or a directory of single-locus alignments” (Bui Quang Minh et al., 2020). Overall, IQtree is extremely fast and mostly accurate (when using the optimal amount of data). However, depending on the amount of data inputted into IQTree (extremely small or extremely large amounts of data), it may not always create the most accurate tree. Another weakness of IQTree is that it primarily uses fast NNI, which can produce less accurate trees than SPR (depending on the data size).
